### PR TITLE
Horizontal bar graphs

### PR DESCRIPTION
**Pull request by cptran777**
Add a horizontal bar chart component and styling in preparation for new metadata health tab.

`ember test` results:
```
ok 1 Chrome 68.0 - ESLint | mirage: mirage/data/schema.js
ok 2 Chrome 68.0 - ESLint | mirage: mirage/factories/access.js
...
ok 420 Chrome 68.0 - Unit | Utility | validators/urn: buildLiUrn
ok 421 Chrome 68.0 - ember-qunit: Ember.onerror validation: Ember.onerror is functioning properly

1..421
# tests 421
# pass  415
# skip  6
# fail  0

# ok
```

Reference:
screen shot 2018-07-30 at 6 15 07 pm

### CODE REVIEW

File: wherehows-web/app/components/visualization/charts/horizontal-bar-chart.ts [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+  BAR_MARGIN_BOTTOM = 8;
+  LABEL_HEIGHT = 15;
+  LABEL_MARGIN_BOTTOM = 16;
+
+  /**
+   * Overall width of our chart. If we have a size, that means that the component and available space
+   * has been measured.
+   * @type {ComputedProperty<number>}
+   */
+  width: ComputedProperty<number> = computed('size', function(this: HorizontalBarChart) {
```
Comment (theseyi) [inline wherehows-web/app/components/visualization/charts/horizontal-bar-chart.ts]: Please annotate types for all computed property callbacks
```typescript
function(this: ...): number {
 // ...
}
```

File: wherehows-web/app/components/visualization/charts/horizontal-bar-chart.ts [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+    return get(this, 'size') ? this.$(this.element).width() : 0;
+  });
+
+  /**
+   * Overall height of our chart, calculated based on the amount of items we have in our series
+   * @type {ComputedProperty<number>}
+   */
+  height: ComputedProperty<number> = computed('categories', function(this: HorizontalBarChart) {
```
Comment (theseyi) [inline wherehows-web/app/components/visualization/charts/horizontal-bar-chart.ts]: Same here, and for all following

File: wherehows-web/app/components/visualization/charts/horizontal-bar-chart.ts [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+
+  /**
+   * Used as a predicate function in the mapping function for the series array to be mapped into the
+   * seriesData array, this function adds values to each chart datum object so that the svg template
+   * can render each bar with the correct dimensions and position
+   * @param this - explicit this keyword declaration for typescript
+   * @param data - single datum object in our series
+   * @param index - current index in the series array
+   */
+  bar(this: HorizontalBarChart, data: IChartDatum, index: number) {
```
Comment (theseyi) [inline wherehows-web/app/components/visualization/charts/horizontal-bar-chart.ts]: Please annotate `bar` with a return type

File: wherehows-web/app/components/visualization/charts/horizontal-bar-chart.ts [hunk 4]
```diff
@@ -123,6 +153,16 @@ export default class HorizontalBarChart extends Component {
     };
   }
 
+  constructor() {
+    super();
```
Comment (theseyi) [inline wherehows-web/app/components/visualization/charts/horizontal-bar-chart.ts]: `super(...arguments);`

Comment (cptran777) [reply wherehows-web/app/components/visualization/charts/horizontal-bar-chart.ts]: Ah interesting. That used to throw a typescript error for me but I guess after the update it works haha.

**Pull request merged**